src/openforms/registrations/base.py

```diff
@@ -29,7 +29,14 @@ class PreRegistrationResult:
     data: dict | None = None
 
 
-class BasePlugin(ABC, AbstractBasePlugin):
+class Options(TypedDict):
+    pass
+
+
+OptionsT = TypeVar("OptionsT", bound=Options)
+
+
+class BasePlugin(Generic[OptionsT], ABC, AbstractBasePlugin):
```

Comment:
Checked and pyright seems to work and enforce the parametrization of the type var to be a typed dict